src/terrain/draw_terrain_raster.js

```diff
@@ -141,11 +142,13 @@ function drawTerrainForGlobe(painter: Painter, terrain: Terrain, sourceCache: So
 
     let program, programMode;
     const showWireframe = painter.options.showTerrainWireframe ? SHADER_TERRAIN_WIREFRAME : SHADER_DEFAULT;
+    const tr = painter.transform;
+    const useCustomAntialiasing = !painter.style.map._antialias && globeToMercatorTransition(tr.zoom) === 0.0;
 
     const setShaderMode = (mode, isWireframe) => {
         if (programMode === mode)
             return;
-        const defines = [shaderDefines[mode], 'PROJECTION_GLOBE_VIEW'];
+        const defines = useCustomAntialiasing ? [shaderDefines[mode], 'CUSTOM_ANTIALIASING', 'PROJECTION_GLOBE_VIEW'] : [shaderDefines[mode], 'PROJECTION_GLOBE_VIEW'];
```

Comment:
To be safe here, we also need to check against: `this.extStandardDerivative` before injecting `CUSTOM_ANTIALIASING`, otherwise the code path might try to call `fwidth` incorrectly.